OsmAnd/src/net/osmand/plus/GPXUtilities.java

```diff
@@ -1758,6 +1822,32 @@ public static GPXFile loadGPXFile(Context ctx, InputStream f) {
 		return res;
 	}
 
+	private static Track parseRoutePointExtension(XmlPullParser parser) {
```

Comment:
Remove parseRoutePointExtension and try to fill segments/points with existing code (line 1683, 1695, etc)